### PR TITLE
Adding support for hipStreamWriteValue32/64 and hipStreamWaitValue32/64

### PR DESCRIPTION
**Pull request by dylanangus**
Adding support for hipStreamWrite32/64 and hipStreamWaitValue32/64 functions.

This PR is dependent upon other PRs and will be rebased once the others have been merged

### CODE REVIEW

**Comment by dylanangus** [comment]
The following tests have been skipped pending bug-fixes
* Unit_hipStreamValue_Wait32_Blocking_Mask_Gte_1
* Unit_hipStreamValue_Wait32_Blocking_Mask_Gte_2
* Unit_hipStreamValue_Wait32_Blocking_Mask_Nor_2
* Unit_hipStreamValue_Wait32_Blocking_NoMask_Gte
* Unit_hipStreamValue_Wait64_Blocking_Mask_Gte_1
* Unit_hipStreamValue_Wait64_Blocking_Mask_Gte_2
* Unit_hipStreamValue_Wait64_Blocking_Mask_Nor_1
* Unit_hipStreamValue_Wait64_Blocking_Mask_Nor_2
* Unit_hipStreamValue_Wait64_Blocking_NoMask_Gte

File: tests/catch/unit/stream/hipStreamValue.cc [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+  SECTION("Registered host memory hipStreamWriteValue32") {
+    INFO("Test writting to registered host pointer using hipStreamWriteValue32");
+    HIP_CHECK(hipStreamWriteValue32(stream, hostPtr32.get(), value32, writeFlag));
+    HIP_CHECK(hipStreamSynchronize(stream));
+    HIP_ASSERT(*hostPtr32 == value32);
+  }
+
+  SECTION("Registered host memory hipStreamWriteValue64") {
+    INFO("Test writting to registered host pointer using hipStreamWriteValue32");
+    HIP_CHECK(hipStreamWriteValue64(stream, hostPtr64.get(), value64, writeFlag));
```
Comment (agunashe) [inline tests/catch/unit/stream/hipStreamValue.cc:110]: Request to please add test cases to verify hipStreamWriteValue64/hipStreamWriteValue32 for arrays. 
Need to test memory with offset

File: tests/catch/unit/stream/hipStreamValue.cc [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+
+  // Allocate Signal Memory
+  int64_t* signalPtr;
+  HIP_CHECK(hipExtMallocWithFlags((void**)&signalPtr, 8, hipMallocSignalMemory));
+
+  std::vector<hipEvent_t> events;
+  initData<isBlocking>(dataPtr.get(), signalPtr, tc, events);
+
+  if (std::is_same<intT, int32_t>::value) {
+    CHECK_HIP_STREAM_VALUE(Write, 32, stream, &(dataPtr.get()[0]), DATA_UPDATE, writeFlag)
```
Comment (agunashe) [inline tests/catch/unit/stream/hipStreamValue.cc:225]: Request to please add test cases to verify hipStreamWaitValue64/hipStreamWaitValue32 for arrays.
Need to test memory with offset

**Review by agunashe — COMMENTED** [review]
Maybe handle arrays with a new PR once this PR gets merged

File: tests/catch/unit/stream/hipStreamValue.cc (hunk 1 above)
Comment (dylanangus) [reply tests/catch/unit/stream/hipStreamValue.cc:110]: Are you requesting for there to be additional tests for array inputs **and** offset inputs? Or offset array inputs?

Comment (dylanangus) [reply tests/catch/unit/stream/hipStreamValue.cc:110]: Additionally, when you mention arrays, are you specifically referencing `hipArray` or just a contiguous piece of memory?

Comment (chrispaquot) [reply tests/catch/unit/stream/hipStreamValue.cc:110]: @agunashe Do you have an answer for Dylan?

Comment (agunashe) [reply tests/catch/unit/stream/hipStreamValue.cc:110]: @dylan-angus-codeplay - need test cases for array contiguous memory as input both for hipStreamWriteValue64/hipStreamWriteValue32 and hipStreamWaitValue64/hipStreamWaitValue32

Comment (fabiomestre) [reply tests/catch/unit/stream/hipStreamValue.cc:110]: @agunashe @chrispaquot I have created an internal ticket to write the tests requested. 
If it's acceptable to you, this PR could still be merged and we will create a new PR for the array tests.

**Review by chrispaquot — APPROVED** [review]

**Pull request merged**